src/hivpy/sexual_behaviour.py

```diff
@@ -51,9 +51,11 @@ def __init__(self, **kwargs):
     # Haven't been able to locate the probabilities for this yet
     # Doing them uniform for now
     def init_sex_behaviour_groups(self, population):
-        population["sex_behaviour"] = np.where(population['sex'] == SexType.Male,
-                                               np.random.choice(MaleSexBehaviour).value,
-                                               np.random.choice(FemaleSexBehaviour).value)
+        for sex in SexType:
+            index = selector(population, sex=(operator.eq, sex))
+            n_sex = sum(index)
+            population.loc[index, "sex_behaviour"] = np.random.choice(
+                SexBehaviours[sex], size=n_sex)
```

Comment:
Out of curiosity, is this so we only generate the values we need or was there something else wrong in the old version?

Reply:
The old version wasn't assigning a unique group for each person, but for each gender would make one random choice out of their groups and assign that to everyone of that gender! Essentially creating an array of copies of a random choice, rather than an array of random choices.